### ISSUE TITLE
deno 1.9.0 bundle (SyntaxError: A class may only have one constructor)

### ISSUE THREAD

**Issue by iugo** [denoland/deno#10252]
with https://deno.land/x/postgres@v0.8.0/mod.ts

ts code is:

```ts
class Query {
  public text: string;
  public args: EncodedArg[];
  public fields?: string[];

  constructor(config: QueryObjectConfig);
  constructor(text: string, ...args: unknown[]);
  //deno-lint-ignore camelcase
  constructor(config_or_text: string | QueryObjectConfig, ...args: unknown[]) {
    let config: QueryConfig;
    if (typeof config_or_text === "string") {
      config = { text: config_or_text, args };
    } else {
      const {
        fields,
        //deno-lint-ignore camelcase
        ...query_config
      } = config_or_text;

      // Check that the fields passed are valid and can be used to map
      // the result of the query
      if (fields) {
        //deno-lint-ignore camelcase
        const clean_fields = fields.map((field) =>
          field.toString().toLowerCase()
        );

        if ((new Set(clean_fields)).size !== clean_fields.length) {
          throw new TypeError(
            "The fields provided for the query must be unique",
          );
        }

        this.fields = clean_fields;
      }

      config = query_config;
    }
    this.text = config.text;
    this.args = this._prepareArgs(config);
  }

  private _prepareArgs(config: QueryConfig): EncodedArg[] {
    const encodingFn = config.encoder ? config.encoder : encode;
    return (config.args || []).map(encodingFn);
  }
}
```
(1.9.0) deno bundle, code is:

```js
class Query {
    text;
    args;
    fields;
    constructor(config2){
    }
    constructor(text, ...args2){
    }
    constructor(config_or_text, ...args1){
        let config1;
        if (typeof config_or_text === "string") {
            config1 = {
                text: config_or_text,
                args: args1
            };
        } else {
            const { fields: fields2 , ...query_config } = config_or_text;
            if (fields2) {
                const clean_fields = fields2.map((field)=>field.toString().toLowerCase()
                );
                if (new Set(clean_fields).size !== clean_fields.length) {
                    throw new TypeError("The fields provided for the query must be unique");
                }
                this.fields = clean_fields;
            }
            config1 = query_config;
        }
        this.text = config1.text;
        this.args = this._prepareArgs(config1);
    }
    _prepareArgs(config) {
        const encodingFn = config.encoder ? config.encoder : encode6;
        return (config.args || []).map(encodingFn);
    }
}
```

should (deno 1.8.2 is):

```js
class Query {
    constructor(config_or_text, ...args1){
        let config1;
        if (typeof config_or_text === "string") {
            config1 = {
                text: config_or_text,
                args: args1
            };
        } else {
            const { fields: fields1 , ...query_config } = config_or_text;
            if (fields1) {
                const clean_fields = fields1.map((field)=>field.toString().toLowerCase()
                );
                if (new Set(clean_fields).size !== clean_fields.length) {
                    throw new TypeError("The fields provided for the query must be unique");
                }
                this.fields = clean_fields;
            }
            config1 = query_config;
        }
        this.text = config1.text;
        this.args = this._prepareArgs(config1);
    }
    _prepareArgs(config) {
        const encodingFn = config.encoder ? config.encoder : encode6;
        return (config.args || []).map(encodingFn);
    }
}
```

**Comment by iugo** [comment]
https://www.typescriptlang.org/play?#code/MYGwhgzhAECKCuBTATgT2gbwFDWsA9gHYQAuy8wJ+yAFAYQGYCWA5gFzSHwC2ARigEoA3DjxFS5StRolEADxIcJTQiwA00AHTawyFhA7xCAa0L4A7oQDaAXWGj6EilVr1mLAPrUPshUrIqLNAAPpw8-Mga2pq6+oYmZpa2ApiiAL5YGUA

```ts
class Query {
  constructor(config: number);
  constructor(text: string, ...args: unknown[]);
  constructor(config_or_text: string | number, ...args: unknown[]) {
  }
}
```

```js
"use strict";
class Query {
    constructor(config_or_text, ...args) {
    }
}
```

**Comment by Liamolucko** [comment]
It looks like this is fixed on `swc` master, so this should be solved by an `swc` upgrade.

**Comment by disyam** [comment]
looks like this is fixed on 1.9.2
